lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -286,74 +295,91 @@ public static function get_theme_data( $deprecated = array(), $options = array()
 
 		}
 
-		if ( ! $options['with_supports'] ) {
+		if ( ! $options['with_supports'] && ! $options['with_block_style_variations'] ) {
 			return static::$theme;
 		}
 
-		/*
-		 * We want the presets and settings declared in theme.json
-		 * to override the ones declared via theme supports.
-		 * So we take theme supports, transform it to theme.json shape
-		 * and merge the static::$theme upon that.
-		 */
-		$theme_support_data = WP_Theme_JSON_Gutenberg::get_from_editor_settings( get_classic_theme_supports_block_editor_settings() );
-		if ( ! wp_theme_has_theme_json() ) {
-			if ( ! isset( $theme_support_data['settings']['color'] ) ) {
-				$theme_support_data['settings']['color'] = array();
-			}
+		$theme_support_data = array(
+			'version'  => WP_Theme_JSON_Gutenberg::LATEST_SCHEMA,
+			'settings' => array(),
+		);
 
-			$default_palette = false;
-			if ( current_theme_supports( 'default-color-palette' ) ) {
-				$default_palette = true;
-			}
-			if ( ! isset( $theme_support_data['settings']['color']['palette'] ) ) {
-				// If the theme does not have any palette, we still want to show the core one.
-				$default_palette = true;
-			}
-			$theme_support_data['settings']['color']['defaultPalette'] = $default_palette;
+		if ( $options['with_supports'] ) {
```

Comment:
The following changes are just to allow the supports and style variations theme data to be opted out of separately.